src/modules/file-transfer-service/FileTransferService.ts

```diff
@@ -120,15 +101,22 @@ export class FileTransferService {
         }
     }
 
-    private getErrorFrom(err: any, fileId: string): Error {
-
-        if (err.isAxiosError) {
-            switch (err.response.status) {
-                case NOT_FOUND:
-                    return new Error(`File ${fileId} not found.`);
+    private handleResponseError(operation: 'upload' | 'metadata retrieval' | 'download' | 'deletion', subject: string):
+        (err: AxiosError) => never {
+        // tslint:disable: max-line-length
+        return (err: AxiosError) => {
+            if (err.isAxiosError) {
```

Comment:
wouldn't the `err` always be of type `AxiosError`?

Reply:
Hmm, not sure, Francisco started this convention and I think it gives us extra safety.

Reply:
This is what docs say: https://github.com/axios/axios#handling-errors

It does not explain what errors you can get but with HTTP which is complex topic I feel more safe with that condition. In fact you made me consider another case where response has not been received. I will make it even safer.

Reply:
Now we will have:

```
            if (err.isAxiosError && err.response != null) {
                switch (err.response.status) {
```

Reply:
i like the extra safety!